packages/dara-core/dara/core/internal/encoder_registry.py

```diff
@@ -114,5 +117,7 @@ def _df_decode_resolver(df: Any):
     pandas.Series: Encoder(serialize=lambda x: x.to_list(), deserialize=lambda x: pandas.Series(x)),
     pandas.Index: Encoder(serialize=lambda x: x.to_list(), deserialize=lambda x: pandas.Index(x)),
     pandas.Timestamp: Encoder(serialize=lambda x: x.isoformat(), deserialize=lambda x: pandas.Timestamp(x)),
-    pandas.DataFrame: Encoder(serialize=lambda x: x.to_json(orient='records'), deserialize=lambda x: _df_decode_resolver(x)),
+    pandas.DataFrame: Encoder(
+        serialize=lambda x: x.to_dict(orient='records'), deserialize=lambda x: _df_decode_resolver(x)
```

Comment:
Won't this output an unserializable dict if e.g. the index is using Timestamp objects? If that's the case perhaps we need to call `jsonable_encoder` on the output of `to_dict` here.